include/mbgl/gfx/gpu_expression.hpp

```diff
@@ -0,0 +1,113 @@
+#pragma once
+
+#include <mbgl/util/color.hpp>
+#include <mbgl/util/variant.hpp>
+
+#include <optional>
+
+#if MLN_DRAWABLE_RENDERER
+
+namespace mbgl {
+namespace style {
+namespace expression {
+class Expression;
+class Interpolate;
+class Step;
+} // namespace expression
+using ZoomCurvePtr = variant<std::nullptr_t, const expression::Interpolate*, const expression::Step*>;
+} // namespace style
+namespace gfx {
+
+enum class GPUInterpType : std::uint16_t {
+    Step,
+    Linear,
+    Exponential,
+    Bezier
+};
+enum class GPUOutputType : std::uint16_t {
+    Float,
+    Color,
+};
+enum class GPUOptions : std::uint16_t {
+    None = 0,
+    IntegerZoom = 1 << 0,
+    Transitioning = 1 << 1,
+};
+
+class GPUExpression;
+using UniqueGPUExpression = std::unique_ptr<const GPUExpression>;
+using MutableUniqueGPUExpression = std::unique_ptr<GPUExpression>;
```

Comment:
Personally I think `std::unique_ptr<GPUExpression> is clearer than `MutableUniqueGPUExpression`(it's also just 4 characters longer), but I understand making custom names for unique pointers is a well-established practice in the codebase at this point. What do you prefer?

Reply:
I just made them all mutable for now.